dht/i_partitioner.cc

```diff
@@ -179,6 +179,16 @@ unsigned shard_of(const schema& s, const token& t) {
     return s.get_sharder().shard_of(t);
 }
 
+static bool token_range_is_empty(const dht::token_range& r) {
+    return r.start()
+        && r.end()
+        && !r.start()->is_inclusive()
+        && !r.end()->is_inclusive()
+        && r.start()->value()._kind == dht::token_kind::key
+        && r.end()->value()._kind == dht::token_kind::key
+        && static_cast<uint64_t>(r.end()->value()._data) - static_cast<uint64_t>(r.start()->value()._data) == 1;
+}
+
```

Comment:
Maybe it would be better to use https://github.com/scylladb/scylla/blob/1b46a0a5f114f50a295be388ceecd808283e87d5/dht/token.hh#L163

and just do:
```
return r.start()
    && r.end()
    && !r.start()->is_inclusive()
    && !r.end()->is_inclusive()
    && r.end().raw() - r.start().raw() < 2;
```

Reply:
The question is whether r can be a wrap-around like (10, 5).

Reply:
`dht::token_range` is `nonwrapping_interval<token>`, so it can't be a wrap-around.
But the problem with raw() is that it collates `after_all_keys` and int64max, which are different things. In particular, (int64max-1, after_all_keys) is definitely not empty, but it fulfils your condition.

Reply:
In practice, it is empty, no? what token can be placed in (int64max -1 , int64max) ?

Reply:
ok I see your point. int64max should be inside (int64max-1, after_all_keys).

Reply:
Then the logic should be:
1. if start is before all keys and r.end().raw() - r.start().raw() == 0 then empty
2. if end is after all keys and r.end().raw() - r.start().raw() == 0 then empty
3. Otherwise if neither start nor end are special r.end().raw() - r.start().raw() < 2
How about something like this?

Reply:
so
```
return r.start()
    && r.end()
    && !r.start()->is_inclusive()
    && !r.end()->is_inclusive()
    && r.end().raw() - r.start().raw() < ((r.start().is_minimum() || r.end().is_maximum()) ? 0 : 2);
```